planner/core/planbuilder.go

```diff
@@ -558,6 +558,19 @@ func (b *PlanBuilder) getPossibleAccessPaths(indexHints []*ast.IndexHint, tblInf
 		}
 
 		hasScanHint = true
+		if hint.IndexNames == nil {
+			// If indexNames is nil, choose TablePath.
+			// For example: `select a from t use index()`.
+			if path := getTablePath(publicPaths); path != nil {
+				if hint.HintType == ast.HintIgnore {
+					ignored = append(ignored, path)
```

Comment:
This ignore case doesn't make sense?

Reply:
Sure. I'll remove this.

Reply:
I think this is another issue. I tried `ignore index()` in MySQL, it returns a syntax Error. 
```
mysql> explain select b from tt ignore index();
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ')' at line 1
```
But TiDB allows such syntax.